### ISSUE TITLE
Default values for generic attributes are unsafe

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#5703]
Currently mypy accepts this unsafe code: 
```python
from typing import Generic, TypeVar, List
  
T = TypeVar('T')

class B(Generic[T]):
    x: List[T] = []

class C1(B[int]):
    pass

class C2(B[str]):
    pass

C1().x.append(42)
C2().x.append('Hm...')
```
A potential solution would be to disallow all default values for generic attributes except `None`. This is however a low priority, since it is a false negative, and a rare corner case.

**Comment by andersk** [comment]
Same issue as #11538.

We don’t need to go so far as disallowing all defaults other than `None`. Any default value is fine as long as the default value has a type that can be written without `T`. For example:

* `x: Sequence[T] = []` is fine because `[]` has type `Sequence[Never]`, which is convertible to `Sequence[T]` by covariance.
* `x: List[T] = []` must be disallowed because no single `[]` may simultaneously have type `List[T]` for all `T`.
